lgc/patch/FragColorExport.cpp

```diff
@@ -431,3 +484,331 @@ Value *FragColorExport::convertToInt(Value *value, bool signedness, Instruction
 }
 
 } // namespace lgc
+
+char LowerFragColorExport::ID = 0;
+
+// =====================================================================================================================
+// Create the color export pass
+ModulePass *lgc::createLowerFragColorExport() {
+  return new LowerFragColorExport();
+}
+
+// =====================================================================================================================
+// Run the lower vertex fetch pass on a module
+//
+// @param [in/out] module : Module
+bool LowerFragColorExport::runOnModule(Module &module) {
+  m_context = &module.getContext();
+  m_pipelineState = getAnalysis<PipelineStateWrapper>().getPipelineState(&module);
+  m_resUsage = m_pipelineState->getShaderResourceUsage(ShaderStageFragment);
+
+  // Gather vertex fetch calls. We can assume they're all in one function, the vertex shader.
+  // We can assume that multiple fetches of the same location, component and type have been CSEd.
+  auto pipelineShaders = &getAnalysis<PipelineShaders>();
+  Function *fragEntryPoint = pipelineShaders->getEntryPoint(ShaderStageFragment);
+  if (fragEntryPoint == nullptr)
+    return false;
+
+  auto retInst = fragEntryPoint->back().getTerminator();
+  auto insertPos = retInst;
```

Comment:
I think the code here should have an IRBuilder, and then pass its reference into subfunctions. That would then allow you to use IRBuilder::CreateIntrinsic for the export intrinsics, rather than emitCall. And you can use IRBuilder methods to get types and constants. I realize you probably moved chunks of code from PatchInOutImportExport, and that code did not use IRBuilder before, but reconstituting it here is a good opportunity to do that upgrade.